front/lib/api/assistant/templates.ts

```diff
@@ -27,19 +27,28 @@ export async function generateMockAgentConfigurationFromTemplate(
     return new Err(new Error("Template not found"));
   }
 
+  const presetModel = {
+    providerId: template.presetProviderId,
+    modelId: template.presetModelId,
+  };
+
+  if (!isSupportedModel(presetModel)) {
+    return new Err(new Error("Unsupported model"));
+  }
+
   return new Ok({
     actions: removeNulls([getAction(template.presetAction)]),
     description: template.description ?? "",
     instructions: template.presetInstructions ?? "",
-    generation: {
-      model: {
-        providerId: template.presetProviderId,
-        modelId: template.presetModelId,
-      } as SupportedModel,
+    model: {
+      ...presetModel,
       temperature:
         ASSISTANT_CREATIVITY_LEVEL_TEMPERATURES[template.presetTemperature],
       forceUseAtIteration: 1,
     },
+    generation: {
+      forceUseAtIteration: 0,
```

Comment:
It was previously set to 1, is it expected?

Reply:
Good catch asked @fontanierh  and it should be null, I added a commit!